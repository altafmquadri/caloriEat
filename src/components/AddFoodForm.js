import React, { Component } from 'react';
import { connect } from 'react-redux'
import moment from 'moment'
import 'react-dates/initialize'
import 'react-dates/lib/css/_datepicker.css'
import { SingleDatePicker } from 'react-dates'
import { addFood } from './actions/food'
import { addMeal } from './actions/meal'

class AddFoodForm extends Component {
    state = {
        date: moment(),
        mealCategory: 'breakfast',
        calendarFocused: false,
    }

    onDateChange = date => {
        if (date) this.setState({ date })
    }

    onSubmit = e => {
        e.preventDefault()
        const food = {
            date: this.state.date,
            mealCategory: this.state.mealCategory
        }
        this.props.addFood(food) //from mapDispatch to props updates the food state, viewing in redux tools shows proper state for food
        const meal = this.props.food //comes from mapStateToProps 
        console.log(this.props, 'logging props before meal assignment');
        console.log(meal, 'in form') //logging this shows it's not the updated state I am expecting
        this.props.addMeal(meal) //trying to add state from food reducer to the meal reducer, meal reducer is not updated via redux tools
    }

    onFocusChange = ({ focused }) => this.setState({ calendarFocused: focused })
    onMealChange = e => this.setState({ mealCategory: e.target.value })

    render() {
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    <select value={this.state.mealCategory}
                        onChange={this.onMealChange}>
                        <option value='breakfast'>Breakfast</option>
                        <option value='lunch'>Lunch</option>
                        <option value='dinner'>Dinner</option>
                        <option value='snack'>Snack</option>
                    </select>
                    <SingleDatePicker
                        date={this.state.date}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                        id="caloriEat-addFood-form" />
                    <button type="submit">Add Food</button>
                </form>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    addFood: food => dispatch(addFood(food)),
    addMeal: meal => dispatch(addMeal(meal))
})

const mapStateToProps = state => ({
    food: state.food
})

export default connect(mapStateToProps, mapDispatchToProps)(AddFoodForm)