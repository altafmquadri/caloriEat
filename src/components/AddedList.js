import React from 'react';
import { connect } from 'react-redux'

const AddedList = (props) => {
    console.log(props);
    return (
        <div>
            {props.list.map(meal => (
                <li key={meal.uuid}>{`${meal.foodItem} added, calories:${meal.calories}`}</li>
            ))}
        </div>
    );
}

const mapStateToProps = (state) => ({
    list: state.list
})

export default connect(mapStateToProps)(AddedList)