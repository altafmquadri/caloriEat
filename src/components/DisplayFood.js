import React, { Component } from 'react';
import { connect } from 'react-redux'
import { fetchServing } from './actions/food'


class DisplayFood extends Component {
    state = {
        quantity: '',
        measureURI: '',
        foodId: ''
    }

    onChange = e => {
        this.setState({
            quantity: 1,
            measureURI: e.target.value,
            foodId: this.props.food.foodId
        }, () => this.props.fetchServing(this.state))
    }

    render() {
        console.log(this.props);
        const displayContents = () => {
            if (this.props.food.foodItem) {
                const { foodItem, calories, carbohydrates, fat, protein, fiber, image, measures } = this.props.food
                const contents = (
                    <div>
                        <h1>{foodItem}</h1>
                        <h3>Calories: {calories}</h3>
                        <h3>Carbohydrates: {carbohydrates}g</h3>
                        <h3>Fat: {fat}g</h3>
                        <h3>Protein: {protein}g</h3>
                        <h3>Fiber: {fiber}g</h3>
                        <img alt={foodItem} src={image}></img>
                        <select
                            onChange={this.onChange}
                            value={this.state.measureURI}>
                            {measures.map((measure, i) => <option key={i} value={measure.uri}>{measure.label}</option>)}
                        </select>
                    </div >
                )
                return contents
            } else return null

        }
        console.log(this.props);
        return (
            <div>
                {displayContents()}
            </div>

        )
    }
}




const mapStateToProps = (state) => ({
    food: state.food
})

const mapPropsToDispatch = (dispatch) => ({
    fetchServing: obj => dispatch(fetchServing(obj))
})

export default connect(mapStateToProps, mapPropsToDispatch)(DisplayFood)