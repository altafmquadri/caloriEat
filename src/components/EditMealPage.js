import React, { Component } from 'react';
import { connect } from 'react-redux'
import { editServing } from './actions/meal';

class EditMealPage extends Component {
    constructor(props) {
        super(props)
        this.state = {
            uuid: this.props.meal.uuid,
            foodId: this.props.meal.foodId,
            measureURI: this.props.meal.measureURI,
            quantity: this.props.meal.quantity,
            mealCategory: this.props.meal.mealCategory
        }
    }

    onMeasureChange = e => (this.setState({ measureURI: e.target.value }))
    onQuantityChange = e => (this.setState({ quantity: e.target.value }))
    onMealCategoryChange = e => (this.setState({ mealCategory: e.target.value }))

    onSubmit = e => {
        e.preventDefault()
        const { foodId, measureURI, quantity, uuid, mealCategory } = this.state
        const editsObj = {
            foodId,
            measureURI,
            quantity: parseInt(quantity)
        }
        this.props.editServing(uuid, editsObj, mealCategory)

    }

    render() {
        const { foodItem, calories, measures } = this.props.meal
        const { measureURI, quantity, mealCategory } = this.state
        return (
            <div>
                <h1>Edit Meal</h1>
                <h3>{foodItem}</h3>
                <h3>{calories}</h3>
                <div>
                    <form onSubmit={this.onSubmit}>
                        <select
                            value={measureURI}
                            onChange={this.onMeasureChange}>
                            {measures.map((measure, i) => <option key={i} value={measure.uri}>{measure.label}</option>)}
                        </select>
                        <input
                            value={quantity}
                            onChange={this.onQuantityChange}
                            type="text"></input>
                        <select value={mealCategory}
                            onChange={this.onMealCategoryChange}>
                            <option value='breakfast'>Breakfast</option>
                            <option value='lunch'>Lunch</option>
                            <option value='dinner'>Dinner</option>
                            <option value='snack'>Snack</option>
                        </select>
                        <button type="submit">Edit Meal</button>
                    </form>
                </div>
            </div>
        )
    }
}


const mapStateToProps = (state, props) => ({
    meal: state.meals.find(meal => meal.uuid === props.match.params.uuid)
})

const mapDispatchToProps = dispatch => ({
    editServing: (uuid, fetchObj, mealCategory) => dispatch(editServing(uuid, fetchObj, mealCategory))
})
export default connect(mapStateToProps, mapDispatchToProps)(EditMealPage)