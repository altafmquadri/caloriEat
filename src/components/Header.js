import React from 'react';
import { NavLink } from 'react-router-dom'

const Header = () => {
    return (
        <div>
            <NavLink to='/'>Home</NavLink>
            <NavLink to='/search'>Search</NavLink>
        </div>
    );
}

export default Header;