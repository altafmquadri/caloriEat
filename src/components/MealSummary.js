import React, { Component } from 'react';
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import moment from 'moment'
import { SingleDatePicker } from 'react-dates'
import MealList from './MealList';
import { capitalize } from '../helpers/helpers'

export class MealSummary extends Component {
    state = {
        date: moment(),
        calendarFocused: false
    }

    componentDidMount() {
        this.setDateOnMount()
    }

    setDateOnMount = () => {
        let date
        if (this.props.location.state) {
            date = moment(this.props.location.state.date)
        }
        this.setState({ date: date || moment() },
            () => this.props.location.state = undefined
        )
    }

    onDateChange = date => {
        if (date) this.setState({ date })
    }
    onFocusChange = ({ focused }) => this.setState({ calendarFocused: focused })

    renderMealCategory = (mealCategory) => {
        return this.props.meals.filter(meal => meal.mealCategory === mealCategory && moment(meal.date).isSame(this.state.date, 'day'))
    }

    renderFilteredTotal = (mealCategory) => {
        return this.props.meals.filter(meal => meal.mealCategory === mealCategory && moment(meal.date).isSame(this.state.date, 'day'))
            .reduce((sum, n) => sum + n.calories, 0)
    }

    renderTotal = () => (this.props.meals.filter(meal => moment(meal.date).isSame(this.state.date, 'day')).reduce((sum, n) => sum + n.calories, 0))

    renderMeals = () => {
        const categories = ['breakfast', 'lunch', 'dinner', 'snack']

        return categories.map(category => {
            return (
                <div key={category}>
                    <h1>{capitalize(category)}</h1>
                    <MealList meals={this.renderMealCategory(category)} date={this.state.date.valueOf()} />
                    <h4>Total: {this.renderFilteredTotal(category)}</h4>
                </div>
            )
        })
    }

    goToToday = () => this.setState({ date: moment() })

    render() {
        return (
            <div>
                <h1>Summary Page</h1>
                <SingleDatePicker
                    date={this.state.date}
                    onDateChange={this.onDateChange}
                    focused={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                    id="caloriEat-meal-summary" />
                {this.state.date.isSame(moment(), 'day') ? null : <button onClick={this.goToToday}>Today</button>}
                {this.renderMeals()}
                <div>
                    <h1>Total Calories Consumed: {this.renderTotal()}</h1>
                </div>
            </div >
        );
    }
}

const mapStateToProps = (state) => ({
    meals: state.meals
})

export default withRouter(connect(mapStateToProps)(MealSummary))