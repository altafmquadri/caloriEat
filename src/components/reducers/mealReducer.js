const mealReducer = (state = [], action) => {
    switch (action.type) {
        case 'ADD_MEAL': {
            const meal = action.payload
            return [...state, meal]
        }
        case 'EDIT_MEAL': {
            const {
                calories,
                totalNutrients: {
                    CHOCDF: { quantity: carbohydrates },
                    FAT: { quantity: fat },
                    FIBTG: { quantity: fiber },
                    PROCNT: { quantity: protein },
                }
            } = action.payload
            const item = {
                ...state.find(meal => meal.uuid === action.uuid),
                ...action.properties,
                calories,
                carbohydrates: parseFloat(carbohydrates).toFixed(2),
                fat: parseFloat(fat).toFixed(2),
                fiber: parseFloat(fiber).toFixed(2),
                protein: parseFloat(protein).toFixed(2),
            }
            return state.map(meal => {
                if (meal.uuid === action.uuid) {
                    return meal = { ...meal, ...item }
                }
                return meal
            })
        }
        case 'DELETE_MEAL': {
            const id = action.payload
            return state.filter(meal => meal.uuid !== id)
        }
        default:
            return state
    }
}

export default mealReducer