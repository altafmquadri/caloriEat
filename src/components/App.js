import React, { Component } from 'react';
import InputForm from './InputForm'
import DisplayFood from './DisplayFood'

class App extends Component {
    render() {
        return (
            <div>
                <InputForm />
                <DisplayFood />
            </div>
        )
    }
}

export default App;