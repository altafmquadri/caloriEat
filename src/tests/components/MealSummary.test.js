import React from 'react';
import moment from 'moment'
import { shallow } from 'enzyme'
import { MealSummary } from '../../components/MealSummary'
import { meals } from '../fixtures/mealReducer'


describe('<MealSummary />', () => {
    test('should render MealSummary where date defaults to today', () => {
        const location = {
            state: undefined
        }
        const wrapper = shallow(<MealSummary meals={meals} location={location} />)
        expect(wrapper.find('button').length).toBe(0)
        expect(wrapper).toMatchSnapshot()
    })
    test('should render MealSummary with date selected', () => {
        const date = moment('2020-08-17T18:37:00.048Z')
        const location = {
            state: undefined
        }
        const wrapper = shallow(<MealSummary meals={meals} location={location} />)
        wrapper.setState({ date })
        expect(wrapper.state('date')).toBe(date)
        expect(wrapper.find('button').text()).toBe('Today')
        expect(wrapper).toMatchSnapshot()
    })
    test('componentDidMount should be called with setDateOnMount', () => {
        const setDateOnMount = jest.fn()
        const location = {
            state: undefined
        }
        const componentDidMount = jest.spyOn(MealSummary.prototype, 'componentDidMount')
        const wrapper = shallow(<MealSummary meals={meals} location={location} setDateOnMount={setDateOnMount} />)
        expect(componentDidMount).toHaveBeenCalledTimes(1)
        // expect(setDateOnMount).toHaveBeenCalledTimes(1) //getting error here
    })
})