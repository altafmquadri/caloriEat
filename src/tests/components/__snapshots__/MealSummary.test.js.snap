// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MealSummary /> should render MealSummary where date defaults to today 1`] = `
<div>
  <h1>
    Summary Page
  </h1>
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    appendToBody={false}
    block={false}
    calendarInfoPosition="bottom"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    dayPickerNavigationInlineStyles={null}
    daySize={39}
    disableScroll={false}
    disabled={false}
    displayFormat={[Function]}
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    horizontalMonthPadding={13}
    id="caloriEat-meal-summary"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepFocusOnInput={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPosition="navPositionTop"
    navPrev={null}
    noBorder={false}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsSelected": [Function],
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month.",
        "jumpToPrevMonth": "Move backward to switch to the previous month.",
        "keyboardBackwardNavigationInstructions": "Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardForwardNavigationInstructions": "Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
        "moveFocusByOneMonth": "Switch months.",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
        "openThisPanel": "Open this panel.",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field.",
        "roleDescription": "datepicker",
        "selectFocusedDate": "Select the date in focus.",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    regular={false}
    renderCalendarInfo={null}
    renderDayContents={null}
    renderMonthElement={null}
    renderMonthText={null}
    renderNavNextButton={null}
    renderNavPrevButton={null}
    renderWeekHeaderElement={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    small={false}
    verticalHeight={null}
    verticalSpacing={22}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <div
    key="breakfast"
  >
    <h1>
      Breakfast
    </h1>
    <MealList
      date={0}
      meals={Array []}
    />
    <h4>
      Total: 
      0
    </h4>
  </div>
  <div
    key="lunch"
  >
    <h1>
      Lunch
    </h1>
    <MealList
      date={0}
      meals={Array []}
    />
    <h4>
      Total: 
      0
    </h4>
  </div>
  <div
    key="dinner"
  >
    <h1>
      Dinner
    </h1>
    <MealList
      date={0}
      meals={Array []}
    />
    <h4>
      Total: 
      0
    </h4>
  </div>
  <div
    key="snack"
  >
    <h1>
      Snack
    </h1>
    <MealList
      date={0}
      meals={Array []}
    />
    <h4>
      Total: 
      0
    </h4>
  </div>
  <div>
    <h1>
      Total Calories Consumed: 
      0
    </h1>
  </div>
</div>
`;

exports[`<MealSummary /> should render MealSummary with date selected 1`] = `
<div>
  <h1>
    Summary Page
  </h1>
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    appendToBody={false}
    block={false}
    calendarInfoPosition="bottom"
    customCloseIcon={null}
    customInputIcon={null}
    date={"2020-08-17T18:37:00.048Z"}
    dayPickerNavigationInlineStyles={null}
    daySize={39}
    disableScroll={false}
    disabled={false}
    displayFormat={[Function]}
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    horizontalMonthPadding={13}
    id="caloriEat-meal-summary"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepFocusOnInput={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPosition="navPositionTop"
    navPrev={null}
    noBorder={false}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsSelected": [Function],
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month.",
        "jumpToPrevMonth": "Move backward to switch to the previous month.",
        "keyboardBackwardNavigationInstructions": "Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardForwardNavigationInstructions": "Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
        "moveFocusByOneMonth": "Switch months.",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
        "openThisPanel": "Open this panel.",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field.",
        "roleDescription": "datepicker",
        "selectFocusedDate": "Select the date in focus.",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    regular={false}
    renderCalendarInfo={null}
    renderDayContents={null}
    renderMonthElement={null}
    renderMonthText={null}
    renderNavNextButton={null}
    renderNavPrevButton={null}
    renderWeekHeaderElement={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    small={false}
    verticalHeight={null}
    verticalSpacing={22}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <button
    onClick={[Function]}
  >
    Today
  </button>
  <div
    key="breakfast"
  >
    <h1>
      Breakfast
    </h1>
    <MealList
      date={1597689420048}
      meals={
        Array [
          Object {
            "calories": 148,
            "carbohydrates": "11.71",
            "date": "2020-08-17T18:36:13.173Z",
            "fat": "7.93",
            "fiber": "0.00",
            "foodId": "food_b49rs1kaw0jktabzkg2vvanvvsis",
            "foodItem": "milk",
            "image": "https://www.edamam.com/food-img/7c9/7c9962acf83654a8d98ea6a2ade93735.jpg",
            "mealCategory": "breakfast",
            "measureURI": "http://www.edamam.com/ontologies/edamam.owl#Measure_cup",
            "measures": Array [
              Object {
                "label": "Serving",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_serving",
              },
              Object {
                "label": "Gram",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_gram",
              },
              Object {
                "label": "Ounce",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_ounce",
              },
              Object {
                "label": "Pound",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_pound",
              },
              Object {
                "label": "Kilogram",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_kilogram",
              },
              Object {
                "label": "Tablespoon",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_tablespoon",
              },
              Object {
                "label": "Cup",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_cup",
              },
              Object {
                "label": "Bottle",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_bottle",
              },
              Object {
                "label": "Fluid ounce",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_fluid_ounce",
              },
              Object {
                "label": "Quart",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_quart",
              },
              Object {
                "label": "Gallon",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_gallon",
              },
            ],
            "protein": "7.69",
            "quantity": 1,
            "uuid": "1",
          },
        ]
      }
    />
    <h4>
      Total: 
      148
    </h4>
  </div>
  <div
    key="lunch"
  >
    <h1>
      Lunch
    </h1>
    <MealList
      date={1597689420048}
      meals={
        Array [
          Object {
            "calories": 255,
            "carbohydrates": "40.76",
            "date": "2020-08-17T18:37:00.048Z",
            "fat": "8.49",
            "fiber": "3.40",
            "foodId": "food_aqa4qtwapci86kar0l275bjqmkal",
            "foodItem": "french fries",
            "image": "https://www.edamam.com/food-img/559/559b143365bf3dde8d5889ec627e1da7.jpg",
            "mealCategory": "lunch",
            "measureURI": "http://www.edamam.com/ontologies/edamam.owl#Measure_ounce",
            "measures": Array [
              Object {
                "label": "Serving",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_serving",
              },
              Object {
                "label": "Strip",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_strip",
              },
              Object {
                "label": "Gram",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_gram",
              },
              Object {
                "label": "Ounce",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_ounce",
              },
              Object {
                "label": "Pound",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_pound",
              },
              Object {
                "label": "Kilogram",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_kilogram",
              },
            ],
            "protein": "3.98",
            "quantity": 6,
            "uuid": "3",
          },
        ]
      }
    />
    <h4>
      Total: 
      255
    </h4>
  </div>
  <div
    key="dinner"
  >
    <h1>
      Dinner
    </h1>
    <MealList
      date={1597689420048}
      meals={Array []}
    />
    <h4>
      Total: 
      0
    </h4>
  </div>
  <div
    key="snack"
  >
    <h1>
      Snack
    </h1>
    <MealList
      date={1597689420048}
      meals={
        Array [
          Object {
            "calories": 55,
            "carbohydrates": "8.52",
            "date": "2020-08-17T18:36:36.867Z",
            "fat": "2.30",
            "fiber": "0.35",
            "foodId": "food_bihs8eobc58jvqbhmt6s3bym1rhf",
            "foodItem": "oreo",
            "image": "https://www.edamam.com/food-img/a90/a9011c79689c79db4f1a64776509dc69.jpg",
            "mealCategory": "snack",
            "measureURI": "http://www.edamam.com/ontologies/edamam.owl#Measure_cookie",
            "measures": Array [
              Object {
                "label": "Whole",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_unit",
              },
              Object {
                "label": "Serving",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_serving",
              },
              Object {
                "label": "Cookie",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_cookie",
              },
              Object {
                "label": "Oreo",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_oreo",
              },
              Object {
                "label": "Sandwich",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_sandwich",
              },
              Object {
                "label": "Box",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_box",
              },
              Object {
                "label": "Container",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_container",
              },
              Object {
                "label": "Package",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_package",
              },
              Object {
                "label": "Gram",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_gram",
              },
              Object {
                "label": "Ounce",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_ounce",
              },
              Object {
                "label": "Pound",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_pound",
              },
              Object {
                "label": "Kilogram",
                "uri": "http://www.edamam.com/ontologies/edamam.owl#Measure_kilogram",
              },
            ],
            "protein": "0.63",
            "quantity": 1,
            "uuid": "2",
          },
        ]
      }
    />
    <h4>
      Total: 
      55
    </h4>
  </div>
  <div>
    <h1>
      Total Calories Consumed: 
      458
    </h1>
  </div>
</div>
`;
